File: ComponentsInJava/app/src/main/java/com/techmania/componentsinjava/MainActivity.java
package com.techmania.componentsinjava;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView text;
    Button button,display;

    EditText name;
    Button ok;
    TextView result;
    String userName;
    ImageView image;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        text = findViewById(R.id.textExample);
        button = findViewById(R.id.doMagic);
        display = findViewById(R.id.buttonDisplay);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                button.setBackgroundColor(Color.BLACK);
                text.setText("I have done my magic.");
                text.setVisibility(View.INVISIBLE);
            }
        });

        text.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                text.setTextColor(Color.BLACK);
                text.setBackgroundColor(Color.YELLOW);
            }
        });

        display.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                text.setVisibility(View.VISIBLE);
            }
        });

        name = findViewById(R.id.editTextName);
        ok = findViewById(R.id.buttonOk);
        result = findViewById(R.id.result);

        ok.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                userName = name.getText().toString();
                result.setText(userName);
            }
        });

        image = findViewById(R.id.imageView);
        image.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                image.setImageResource(R.drawable.pic2);
            }
        });
    }
}

File: FragmentLifecycle/app/src/main/java/com/example/fragmentlifecycle/MainActivity.kt
package com.example.fragmentlifecycle

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Log.d("Message","First Activity onCreate")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("Message","First Activity onDestroy")
    }

    override fun onPause() {
        super.onPause()
        Log.d("Message","First Activity onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.d("Message","First Activity onStop")
    }

    override fun onStart() {
        super.onStart()
        Log.d("Message","First Activity onStart")
    }

    override fun onResume() {
        super.onResume()
        Log.d("Message","First Activity onResume")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d("Message","First Activity onRestart")
    }
}